src/Nethermind/Nethermind.Optimism/CL/P2P/OptimismCLP2P.cs

```diff
@@ -22,40 +22,44 @@
 using ILogger = Nethermind.Logging.ILogger;
 using Nethermind.Merge.Plugin.Data;
 using Nethermind.Optimism.CL;
-using Nethermind.Optimism.Rpc;
-using Nethermind.Core.Crypto;
-using Nethermind.JsonRpc;
 using Snappier;
 using Nethermind.Libp2p;
 using Nethermind.Libp2p.Protocols.PubsubPeerDiscovery;
+using Channel = System.Threading.Channels.Channel;
 
 namespace Nethermind.Optimism;
 
 public class OptimismCLP2P : IDisposable
 {
+    private const int MaxGossipSize = 10485760;
+
     private readonly ServiceProvider _serviceProvider;
-    private PubsubRouter? _router;
-    private readonly CancellationTokenSource _cancellationTokenSource = new();
     private readonly ILogger _logger;
-    private readonly IOptimismEngineRpcModule _engineRpcModule;
+    private readonly IExecutionEngineManager _executionEngineManager;
     private readonly P2PBlockValidator _blockValidator;
     private readonly Multiaddress[] _staticPeerList;
     private readonly ICLConfig _config;
-    private ILocalPeer? _localPeer;
-    private readonly Task _mainLoopTask;
-
     private readonly string _blocksV2TopicId;
+    private readonly Channel<ExecutionPayloadV3> _blocksP2PMessageChannel = Channel.CreateBounded<ExecutionPayloadV3>(10); // for safety add capacity
 
+    private PubsubRouter? _router;
+    private ILocalPeer? _localPeer;
     private ITopic? _blocksV2Topic;
+    private ulong _headPayloadNumber;
```

Comment:
Except for `_blocksV2Topic` which requires disposing, these can become local variables.